### ISSUE TITLE
Configuration problems with the latest v1.1.1 of astropy

### ISSUE THREAD

**Issue by mishinma** [tardis-sn/tardis#483]
So with the version 1.1.1 of astropy all tests raise ValueError:
```
/home/misha/.virtualenvs/tardis/local/lib/python2.7/site-packages/astropy/tests/pytest_plugins.py:458: in pytest_runtest_setup
    config_dir = item.config.getini('config_dir')
_pytest.config:773: in getini
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_pytest.config.Config object at 0x7f76a5349450>, name = 'config_dir'

>   ???
E   ValueError: unknown configuration value: u'config_dir'

```
The problem most likely arises from the changed implementation of the `pytest_runtest_setup` hook in 
astropy/tests/pytest_plugins.py:

```
def pytest_runtest_setup(item):
    config_dir = item.config.getini('config_dir')
    cache_dir = item.config.getini('cache_dir')

    # Command-line options can override, however
    config_dir = item.config.getoption('config_dir') or config_dir
    cache_dir = item.config.getoption('cache_dir') or cache_dir

    # We can't really use context managers directly in py.test (although
    # py.test 2.7 adds the capability), so this may look a bit hacky
    if config_dir:
        item.set_temp_config = set_temp_config(config_dir)
        item.set_temp_config.__enter__()
    if cache_dir:
        item.set_temp_cache = set_temp_cache(cache_dir)
        item.set_temp_cache.__enter__()

    # Store a list of the currently opened files so we can compare
    # against them when the test is done.
    if item.config.getvalue('open_files'):
        item.open_files = _get_open_file_list()

    if ('remote_data' in item.keywords and
            not item.config.getvalue("remote_data")):
        pytest.skip("need --remote-data option to run")
```
I think registering these names in the conftest.py should solve the problem. Will try it out.

**Comment by unoebauer** [comment]
Can confirm this issue. It goes away when downgrading to at least astropy v.1.0.8.

**Comment by mishinma** [comment]
From the [changelog ](http://docs.astropy.org/en/stable/changelog.html)of the v1.1.1:
> Added new test config options, config_dir and cache_dir (these can be edited in setup.cfg or as extra command-line options to py.test) for setting the locations to use for the Astropy config files and download caches (see also the related set_temp_config/cache features added in astropy.config)

Also:

> Added new tools set_temp_config and set_temp_cache which can be used either as function decorators or context managers to temporarily use alternative directories in which to read/write the Astropy config files and download caches respectively. This is especially useful for testing, though set_temp_cache may also be used as a way to provide an alternative (application specific) download cache for large data files, rather than relying on the default cache location in users’ home directories.

Is it needed in Tardis?
As a current solution to the issue I suggest overriding `pytest_runtest_setup` from the astropy/tests/pytest_plugins.py with the hook from the old version:

```
def pytest_runtest_setup(item):
    # Store a list of the currently opened files so we can compare
    # against them when the test is done.
    if item.config.getvalue('open_files'):
        item.open_files = _get_open_file_list()

    if ('remote_data' in item.keywords and
            not item.config.getvalue("remote_data")):
        pytest.skip("need --remote-data option to run")
```
Tests run fine, I can make a PR.

**Comment by unoebauer** [comment]
@wkerzendorf, @orbitfold 
Thoughts on this? I'd be fine with using the old pytest_plugins.py hook

**Comment by yeganer** [comment]
What's the problem with just definig those variables in setup.cfg?
Executing the tests in it's own subdirectory and caching data there might
be a good idea. We should consider using the "download additional data"
feature from astropy anyway since we don't want to increase the size of the
repository

Ulrich Noebauer <notifications@github.com> schrieb am Do., 11. Feb. 2016 um
15:23 Uhr:

**Comment by mishinma** [comment]
@yeganer 
Defining them is no problem, but it will be misleading if we don't use them. And to use them the version of astropy in the requirements must be changed to v1.1.1.

**Comment by wkerzendorf** [comment]
I haven't fully looked at the problem. Here is what I can tell you: We might want to talk to the astropy people about solutions to this problem. I think, I "solved" a few things by overriding several functions in "https://github.com/tardis-sn/tardis/blob/master/tardis/conftest.py". 

The package is modeled after the package-template from astropy. We should try to mirror what they do there if possible (https://github.com/astropy/package-template)

**Comment by mishinma** [comment]
@yeganer 
Maybe we can register them in `pytest_addoption` and add something like todo?

**Comment by yeganer** [comment]
This got fixed with #581.
